### PR TITLE
Tree wide various cleanup

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: src/basic/btrfs-util.c [hunk 1]
```diff
@@ -1881,7 +1877,6 @@ int btrfs_subvol_auto_qgroup_fd(int fd, uint64_t subvol_id, bool insert_intermed
                 uint64_t lowest = 256, new_qgroupid;
                 bool created = false;
                 int i;
-
```
Comment (yuwata) [inline src/basic/btrfs-util.c]: This `i` can be also dropped.

Please do not drop this blank line.

File: src/libsystemd-network/sd-dhcp-client.c [hunk 2]
```diff
@@ -643,11 +643,7 @@ int sd_dhcp_client_add_vendor_option(sd_dhcp_client *client, sd_dhcp_option *v)
         assert_return(client, -EINVAL);
         assert_return(v, -EINVAL);
 
-        r = ordered_hashmap_ensure_allocated(&client->vendor_options, &dhcp_option_hash_ops);
-        if (r < 0)
-                return -ENOMEM;
-
-        r = ordered_hashmap_put(client->vendor_options, v, v);
+        r = ordered_hashmap_ensure_put(&client->vendor_options, &dhcp_option_hash_ops, v, v);
```
Comment (yuwata) [inline src/libsystemd-network/sd-dhcp-client.c]: This is not your fault in this PR, but please use `ordered_set` instead.
Then, please also update `client_append_common_discover_request_options()` and `dhcp_option_append()`.

File: src/udev/udev-rules.c [hunk 3]
```diff
@@ -2147,19 +2147,17 @@ static int udev_rule_apply_token_to_event(
...
                 if (!cmd)
                         return log_oom();
 
-                r = ordered_hashmap_put(event->run_list, cmd, token->data);
-                if (r < 0)
+                r = ordered_hashmap_ensure_put(&event->run_list, NULL, cmd, token->data);
+                if (r == -ENOMEM)
                         return log_oom();
+                if (r < 0)
+                        return r;
```
Comment (yuwata) [inline src/udev/udev-rules.c]: Please add a log message:
`return log_rule_error_errno(dev, rules, r, "Failed to store command '%s': %m", cmd);`

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/basic/btrfs-util.c (hunk 1 above)
Comment (ssahani) [reply src/basic/btrfs-util.c]: i is used in 2 loops .

File: src/libsystemd-network/sd-dhcp-client.c (hunk 2 above)
Comment (ssahani) [reply src/libsystemd-network/sd-dhcp-client.c]: I will drop this and open a new PR . thanks

File: src/basic/btrfs-util.c (hunk 1 above)
Comment (yuwata) [reply src/basic/btrfs-util.c]: Why do not you declare `i` in each loop?

Comment (ssahani) [reply src/basic/btrfs-util.c]: that can be done yeah

**Comment by ssahani** [comment]
updated. thanks for the review. ptal

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**